From b092172e56393fad70e472afa26df4f886dfdbc0 Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Fri, 24 Aug 2012 16:19:45 +0000
Subject: build: fix for "make -j" safety

A make rule with multiple (non-pattern) targets just says that each of
those files can be built by the rule, not that the rule builds all of
them at once. So under "make -j", extensions/Makefile would run three
copies of glib-ginterface-gen.py at once, which could end up deleting
each others' files and causing a make failure. Fix.

Signed-off-by: Jonny Lamb <jonny.lamb@collabora.co.uk>
---
diff --git a/extensions/Makefile.am b/extensions/Makefile.am
index 24efb6e..bb837da 100644
--- a/extensions/Makefile.am
+++ b/extensions/Makefile.am
@@ -42,7 +42,10 @@ extensions.html: _gen/all.xml $(tools_dir)/doc-generator.xsl
 		$(tools_dir)/doc-generator.xsl \
 		$< > $@
 
-_gen/svc.c _gen/svc.h _gen/svc-gtk-doc.h: _gen/all.xml \
+_gen/svc.h: _gen/svc.c
+_gen/svc-gtk-doc.h: _gen/svc.c
+
+_gen/svc.c: _gen/all.xml \
 	$(tools_dir)/glib-ginterface-gen.py
 	$(AM_V_GEN)$(PYTHON) $(tools_dir)/glib-ginterface-gen.py \
 		--filename=_gen/svc --signal-marshal-prefix=_idle_ext \
--
cgit v0.9.0.2-2-gbebe
